File: TwoSums.py


class Solution(object):
    
    def two_sum(nums,target):
        mapa={}
        diff=0
        #this is more a mathemathical problem beacuse we need to rest the target
        #to very number in the array and store it in the map, if the diff is in the map
        #we will return de index of that value searching with the key and the actual index of
        #the array i
        for i,n in enumerate(nums):
            diff=target-n
            if diff in mapa:
                return [mapa[diff],i]
            mapa[n]=i
            
            
print(Solution.two_sum([2,1,5,3],4))
                

File: typescript/twoSums.ts
class SolutionTwoSum{
    static twoSum(nums: number[], target: number):number[]{
        const map={};
        let diff:number=0;

        for (let i:number=0;i< nums.length;i++){
            const n= nums[i];
            diff = target -n;
            if(diff in map){
                return [map[diff],i];
            }
            map[n]=i;
        }
        return [0,0];
    };
};

// Example usage:
const solution2 = new SolutionTwoSum();
console.log(SolutionTwoSum.twoSum([2, 7, 11, 15], 9)); // Output: [0, 1]